### PR TITLE
Make FrozenUnderFog footprint include '_'

### LINKED ISSUE #11247: TD crash: maps which contain farmer's field husks
When I start a map which contains a destroyed celery field (v17.husk), it throws an exception and crashes.

It doesn't seem that this issue affects any maps which are included with the game, but it does affect a couple maps I've made. (Including Understanding Power and Media Control).

EDIT: Seems to also apply to all the other field husks: v14 -> v18

```
Tiberian Dawn Mod at Version release-20160508
on map 98edd263db7c925ca19465f5c908c3f80711df70 (Understanding Power by Psydev).
Operating System: Windows (Microsoft Windows NT 6.1.7601 Service Pack 1)
Runtime Version: .NET CLR 4.0.30319.17929
Exception of type `System.ArgumentException`: This frozen actor has no footprint.
Actor Name: v17.husk
Actor Location: 30,1
Input footprint: []
Input footprint (after shroud.Contains): []
   at OpenRA.Traits.FrozenActor..ctor(Actor self, PPos[] footprint, Shroud shroud, Boolean startsRevealed)
   at OpenRA.Mods.Common.Traits.FrozenUnderFog.<Created>c__AnonStorey1.<>m__0(Player player, Int32 playerIndex)
   at OpenRA.Primitives.PlayerDictionary`1..ctor(World world, Func`3 valueFactory)
   at OpenRA.Mods.Common.Traits.FrozenUnderFog.Created(Actor self)
   at OpenRA.World.CreateActor(Boolean addToWorld, String name, TypeDictionary initDict)
   at OpenRA.Mods.Common.Traits.SpawnMapActors.WorldLoaded(World world, WorldRenderer wr)
   at OpenRA.World.LoadComplete(WorldRenderer wr)
   at OpenRA.Game.StartGame(String mapUID, WorldType type)
   at OpenRA.Network.UnitOrders.ProcessOrder(OrderManager orderManager, World world, Int32 clientId, Order order)
   at OpenRA.Network.OrderManager.TickImmediate()
   at OpenRA.Sync.<CheckSyncUnchanged>c__AnonStorey0.<>m__0()
   at OpenRA.Sync.CheckSyncUnchanged[T](World world, Func`1 fn)
   at OpenRA.Sync.CheckSyncUnchanged(World world, Action fn)
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager)
   at OpenRA.Game.LogicTick()
   at OpenRA.Game.Loop()
   at OpenRA.Game.Run()
   at OpenRA.Program.Run(String[] args)
   at OpenRA.Program.Main(String[] args)
```

### PR DESCRIPTION
**Pull request by reaperrr**
Fixes #11247.

### CODE REVIEW

**Comment by pchote** [comment]
Wouldn't it make more sense for frozen actor footprints to include `_` cells?

**Comment by reaperrr** [comment]
Updated.

File: OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs [hunk 1]
```diff
@@ -30,6 +30,10 @@ public static IEnumerable<CPos> Tiles(Ruleset rules, string name, BuildingInfo b
 				footprint = footprint.Concat(new char[dim.X]);
 			}
 
+			// Don't exclude '_' for FrozenUnderFog.
+			if (includePassable)
+				return TilesWhere(name, dim, footprint.ToArray(), a => !char.IsWhiteSpace(a)).Select(t => t + topLeft);
```
Comment (GraionDilach) [inline OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs]: You shouldn't need the `IsWhiteSpace` check by that point, you can simply set a `true` function. Whitespaces are removed in L24 afterall.

**Comment by reaperrr** [comment]
Updated.

**Comment by GraionDilach** [comment]
:+1:

File: OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs [hunk 2]
```diff
@@ -30,6 +30,10 @@ public static IEnumerable<CPos> Tiles(Ruleset rules, string name, BuildingInfo b
 				footprint = footprint.Concat(new char[dim.X]);
 			}
 
+			// Don't exclude '_' for FrozenUnderFog.
+			if (includePassable)
+				return TilesWhere(name, dim, footprint.ToArray(), a => true).Select(t => t + topLeft);
+
 			return TilesWhere(name, dim, footprint.ToArray(), a => a != '_').Select(t => t + topLeft);
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Buildings/FootprintUtils.cs]:
```csharp
return TilesWhere(name, dim, footprint.ToArray(), a => includePassable || a != '_').Select(t => t + topLeft);
```

would make this one line again without a `a => true` duplicate.

**Comment by reaperrr** [comment]
Updated.

**Comment by Mailaender** [comment]
Fix confirmed. :white_check_mark:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/43f89057d894231ed30e09f45d071e6587882879...b529f3d0a439559ec1b3f674a677f1ebc0cd1cb0)

**Pull request merged**